mgmtsystem/hawkular.py

```diff
@@ -691,12 +691,15 @@ def list_operation_definition(self, feed_id, resource_type_id):
         return operations
 
     def list_server_datasource(self, feed_id=None):
-        """Returns list of datasources.
+        """Returns list of datasources (both XA and non XA).
 
          Args:
              feed_id: Feed id of the datasource (optional)
         """
         resources = self.list_resource(feed_id=feed_id, resource_type_id='Datasource')
+        resources.extend(self.list_resource(
+            feed_id=feed_id,
+            resource_type_id='XA Datasource'))
```

Comment:
What happens if you run this method against endpoint that does not support XA Datasource yet?